### ISSUE TITLE
oci_compute_image_shape_compatibility_entry not updating shape

### ISSUE THREAD

**Issue by romuloslv** [oracle/oci-ansible-collection#209]
## Issue Report

**Describe the issue**

When i try to update the shape of a custom image no changes are made.

**Expected behavior**

An update(addition)the new shape to the compatible shapes in the image.

**Environment**

* OS version:

Ubuntu 20.10 5.15.49-linuxkit

* Ansible version:

ansible 2.13.7
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Jun  2 2021, 10:49:15) [GCC 10.3.0]

* OCI Python SDK version:

2.92.0

* OCI Ansible Modules version:

4.15.0

**Ansible playbook to reproduce the issue**

PLAYBOOK

Note: The image shape used was listed as available shape:
oci compute shape list -c ocid1.compartment.oc1..xxxxxxxxxxxxxxxxxxxxxxxxxx --query 'data[*]."shape"' --all

```
[
  ...
  "VM.Standard3.Flex",
  ...
]
```

```
- hosts: localhost
  collections:
    - oracle.oci
  connection: local
  gather_facts: false
  tasks:
    - name: Add shape to imported image
      oci_compute_image_shape_compatibility_entry:
        image_id: ocid1.image.oc1.iad. xxxxxxxxxxxxxxxxxxxxxxxxxx
        shape_name: "VM.Standard3.Flex"
        state: present
      register: result_imp

    - debug:
        msg: "{{ result_imp }}"
```

OUTPUT

```
ok: [localhost] => {
    "changed": false,
    "image_shape_compatibility_entry": {},
    "invocation": {
        "module_args": {
            "api_user": null,
            "api_user_fingerprint": null,
            "api_user_key_file": null,
            "api_user_key_pass_phrase": null,
            "auth_purpose": null,
            "auth_type": "api_key",
            "cert_bundle": null,
            "config_file_location": null,
            "config_profile_name": null,
            "image_id": "ocid1.image.oc1.iad.xxxxxxxxxxxxxxxxxxxxxxxxxx",
            "memory_constraints": null,
            "ocpu_constraints": null,
            "region": "us-ashburn-1",
            "shape_name": "VM.Standard3.Flex",
            "state": "present",
            "tenancy": null
        }
    },
    "oci_ansible_logs": [
        "2023-03-13,18:06:33 DEBUG config.py:195 Config file found at ~/.oci/config",
        "2023-03-13,18:06:33 DEBUG base_client.py:331 No circuit breaker strategy enabled!",
        "2023-03-13,18:06:33 DEBUG base_client.py:348 Endpoint: https://iaas.us-ashburn-1.oraclecloud.com/20160918",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:43 Python Version: 3.8.10",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:33 OCI-Python-Sdk version: 2.92.0",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:52 Ansible Version: 2.13.7",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:61 OCI-Ansible-Collections installed Version: 4.15.0",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:94 OCI-Python-Sdk path: ['/usr/local/lib/python3.8/dist-packages/oci']",
        "2023-03-13,18:06:33 DEBUG oci_version_utils.py:104 Ansible module python path: ['/tmp/ansible_oci_compute_image_shape_compatibility_entry_payload_xq6qqhsa/ansible_oci_compute_image_shape_compatibility_entry_payload.zip/ansible', '/usr/local/lib/python3.8/dist-packages/ansible']",
        "2023-03-13,18:06:33 INFO base_client.py:565  2023-03-13 18:06:33.935291: Request: GET https://iaas.us-ashburn-1.oraclecloud.com/20160918/images/ocid1.image.oc1.iad.xxxxxxxxxxxxxxxxxxxxxxxxxx/shapes/VM.Standard3.Flex",
        "2023-03-13,18:06:33 DEBUG connectionpool.py:1006 Starting new HTTPS connection (1): iaas.us-ashburn-1.oraclecloud.com:443",
        "2023-03-13,18:06:34 DEBUG oci_log_utils.py:84 send: b'GET /20160918/images/ocid1.image.oc1.iad.xxxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxx==\",version=\"1\"\\r\\n\\r\\n'",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 reply: 'HTTP/1.1 404 Not Found\\r\\n'",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: Date: Mon, 13 Mar 2023 18:06:35 GMT",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: opc-request-id: A62338B8044C4A54A79916EE0BCF8069/45C9558069BBF950762EB8B2F4B4F9B7/8C53C790C32DFA9F9FACC9171A197241",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: Access-Control-Expose-Headers: opc-previous-page,opc-next-page,opc-client-info,ETag,opc-total-items,opc-request-id,Location,opc-work-request-id",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: Connection: keep-alive",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: Content-Type: application/json",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: X-Content-Type-Options: nosniff",
        "2023-03-13,18:06:35 DEBUG oci_log_utils.py:84 header: Content-Length: 111",
        "2023-03-13,18:06:35 DEBUG connectionpool.py:461 https://iaas.us-ashburn-1.oraclecloud.com:443 \"GET /20160918/images/ocid1.image.oc1.iad.xxxxxxxxxxxxxxxxxxxxxxxxxx/shapes/VM.Standard3.Flex HTTP/1.1\" 404 111",
        "2023-03-13,18:06:35 DEBUG base_client.py:594  2023-03-13 18:06:35.191375: time elapsed for request A62338B8044C4A54A79916EE0BCF8069: 1.2555352089984808",
        "2023-03-13,18:06:35 DEBUG base_client.py:609  2023-03-13 18:06:35.196028: Response status: 404",
        "2023-03-13,18:06:35 DEBUG base_client.py:873  2023-03-13 18:06:35.198209: python SDK time elapsed for deserializing: 6.741599645465612e-05",
        "2023-03-13,18:06:35 DEBUG oci_resource_utils.py:902 is update necessary for image_shape_compatibility_entry: False"
    ]
}
```

**Comment by dineshsuthar786** [comment]
@romuloslv Thanks for reporting the issue. We are checking on this.

**Comment by dineshsuthar786** [comment]
@romuloslv We have fixed the issue in latest release v4.18.0. Kindly check and let us know if you still face the issue.

**Comment by romuloslv** [comment]
I'm closing this issue, it's ok now guys. Thanks for the support @dineshsuthar786 @nikhiban.
